### PR TITLE
Better password security estimation

### LINKED ISSUE #3378: Password strength estimator
When entering a new password, the user should be shown a visual indicator of the estimated strength of the password.

We could use the following solution:
https://github.com/dropbox/zxcvbn

I would also like to remove the existing password security settings (register_password_*), as these do not always lead to better passwords.

### PR DESCRIPTION
**Pull request by TimWolla**
This thing needs a facelift, but it's functional.

Bildschirmfoto vom 2020-07-16 15-17-37
zxcvbn

------

Resolves #3378

### CODE REVIEW

**Comment by dtdesign** [comment]
@TimWolla Can you please look into the localization of the feedback, it's currently using hardcoded phrases. Since there is no visible development in the past years, it should be safe to fork it and add some i18n capabilities.

**Comment by TimWolla** [comment]
@dtdesign Okay, the feedback from zxcvbn is now being localized, but it is not yet shown anywhere.

**Comment by dtdesign** [comment]
I will take care of it, thanks!

**Comment by dtdesign** [comment]
@TimWolla Please review my changes to see if you spot any issues with it.

File: wcfsetup/install/files/style/layout/form.scss [hunk 1]
```diff
@@ -283,3 +283,66 @@ input {
...
+		left: 0;
+		position: absolute;
+		top: 0;
+		transition: background-color .12s linear, width .12s linear;
+		width: 0;
+	}
+	
+	&[data-score="0"]::before {
+		background-color: #DD2C00;
+		width: 20%;
```
[the remark this answers is not in the document]
Comment (dtdesign) [reply wcfsetup/install/files/style/layout/form.scss]: I've modified the bar sizes to 5/20/50/85/100%. While I do understand your rationale of setting it at 3px or similar, it is too small and could easily be overlooked. 5% is still really small, but noticable, providing a stronger visual feedback of "something is happening".

[the remark this answers is not in the document]
Comment (TimWolla) [reply wcfsetup/install/files/style/layout/form.scss]: I've tested this, I am fine with the new behavior.

**Comment by TimWolla** [comment]
During my retesting with the new bars I noticed the following visual glitch in the RegisterForm:

image

It happens if you focus the password input and immediately unfocus without entering anything. Frankly the Ajax validation within the RegisterForm could / should simply be removed. No other form shows this kind of behavior.

**Comment by dtdesign** [comment]
@TimWolla Can you please create a follow-up issue for this?

**Comment by TimWolla** [comment]
Sure: https://github.com/WoltLab/WCF/issues/3497

**Pull request merged**